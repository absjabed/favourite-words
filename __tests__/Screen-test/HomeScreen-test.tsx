import 'react-native';
import React from 'react';
import Home from '../../src/screens/HomeScreen';
import renderer, { create } from 'react-test-renderer';

it('Home Screen renders correctly', () => {
  renderer.create(<Home />);
});

test('Home Screen snapshot', () => {
  const tree = create(<Home/>);
  expect(tree).toMatchSnapshot();
});

