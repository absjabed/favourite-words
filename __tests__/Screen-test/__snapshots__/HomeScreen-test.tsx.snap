// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#121212",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        },
        Object {
          "backgroundColor": "#121212",
          "justifyContent": "center",
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#0d0d0d",
          "borderBottomColor": "#acacac",
          "borderBottomLeftRadius": 7,
          "borderLeftColor": "#acacac",
          "borderTopColor": "#acacac",
          "borderTopLeftRadius": 7,
          "borderWidth": 0.8,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Ubuntu-Regular",
            "fontSize": 13,
            "fontWeight": "600",
            "justifyContent": "center",
          }
        }
      >
        Search Dictionary
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#1f1f1f",
          "borderBottomColor": "#acacac",
          "borderBottomRightRadius": 7,
          "borderRightColor": "#acacac",
          "borderTopColor": "#acacac",
          "borderTopRightRadius": 7,
          "borderWidth": 0.8,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9b9b9b",
            "fontFamily": "Ubuntu-Regular",
            "fontSize": 13,
            "fontWeight": "100",
            "justifyContent": "center",
          }
        }
      >
        View Favourites
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "padding": 20,
            "paddingHorizontal": 30,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#121212",
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <Modal
          animationType="none"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00000040",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "space-around",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderRadius": 10,
                  "display": "flex",
                  "height": 100,
                  "justifyContent": "space-around",
                  "width": 100,
                }
              }
            >
              <ActivityIndicator
                animating={false}
                color="#829BF8"
                hidesWhenStopped={true}
                size="small"
              />
            </View>
          </View>
        </Modal>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "marginTop": -5,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "marginLeft": 25,
                    "paddingLeft": 3,
                    "width": "80%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "borderColor": "#5f5f5f",
                      "borderRadius": 7,
                      "borderWidth": 1,
                      "height": 40,
                      "marginLeft": 20,
                      "marginRight": 20,
                      "marginTop": 10,
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    blurOnSubmit={false}
                    onChangeText={[Function]}
                    placeholder="Search Dictionary"
                    placeholderTextColor="#9b9b9b"
                    rejectResponderTermination={true}
                    showSoftInputOnFocus={false}
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontFamily": "Ubuntu-Regular",
                        "opacity": 100,
                        "paddingHorizontal": 15,
                        "textAlign": "auto",
                      }
                    }
                    textAlign="left"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#555CC4",
                      "borderRadius": 7,
                      "color": "#ffffff",
                      "marginLeft": 20,
                      "marginRight": 20,
                      "marginTop": 16,
                      "opacity": 1,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontFamily": "Ubuntu-Regular",
                      }
                    }
                  >
                    Search
                  </Text>
                </View>
              </View>
              <Icon
                color="#9b9b9b"
                name="search"
                size={20}
                style={
                  Object {
                    "bottom": 15,
                    "position": "absolute",
                    "right": 135,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "column",
                "paddingTop": 15,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#acacac",
                  "fontFamily": "Ubuntu-Bold",
                  "fontSize": 16,
                }
              }
            >
              Search Results
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Ubuntu-Bold",
                    "fontSize": 24,
                    "textDecorationLine": "underline",
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#acacac",
                    "fontFamily": "Ubuntu-Bold",
                    "fontSize": 15,
                    "paddingLeft": 5,
                  }
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <RCTScrollView
              ListEmptyComponent={
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingTop": "50%",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#9b9b9b",
                        "fontFamily": "Ubuntu-Bold",
                        "fontSize": 18,
                      }
                    }
                  >
                    No Info Available
                  </Text>
                </View>
              }
              contentContainerStyle={
                Object {
                  "padding": 10,
                }
              }
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <View>
                <View
                  onLayout={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingTop": "50%",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#9b9b9b",
                        "fontFamily": "Ubuntu-Bold",
                        "fontSize": 18,
                      }
                    }
                  >
                    No Info Available
                  </Text>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Image
    resizeMode="stretch"
    source={
      Object {
        "testUri": "../../../assets/images/wave.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "borderWidth": 1,
        "height": 90,
        "margin": 5,
        "opacity": 0.4,
        "width": "100%",
        "zIndex": 10,
      }
    }
  />
</View>
`;
